contrib/enum_metrics/enum_metrics/errant_ttl.py

```diff
@@ -0,0 +1,120 @@
+import sys
+import argparse
+import json
+import dbclient as db
+import esclient as es
+import config as cf
+
+LOCALHOST = 'localhost'
+ERRANT_TTL_IN_SECONDS = 259200  # 3 days
+isVerbose = False
+
+def parse_arguments(args):
+    """Parses the supplied arguments"""
+
+    parser = argparse.ArgumentParser(prog="errant_ttl.py", description='show and repair cassandra records with errant ttl, e.g. null')
+
+    subparsers = parser.add_subparsers(help='commands')
+
+    show_errant_ttl = subparsers.add_parser('show', help='get all records in a table with errant ttl')
+    repair_errant_ttl = subparsers.add_parser('repair', help='repair all records in a table with errant ttl')
+    repair_errant_ttl.add_argument('--dryrun', action='store_true')
+
+    for p in [show_errant_ttl, repair_errant_ttl]:
+        p.add_argument('-e', '--env', choices=cf.Config.get_environments(),
+                           default=LOCALHOST, help='Environment we are pointing to')
+        p.add_argument('-cf', '--columnFamily',
+                           required=True, help='name of column family (table) to retrieve rows from')
+        p.add_argument("-v", "--verbose", action="store_true")
+        p.add_argument("-l", "--limit", required=False, help='limit number of records to show or repair')
+
+    args = parser.parse_args()
+
+    global isVerbose
+    if args.verbose: isVerbose = True
+    if isVerbose: print "args: " + str(args)
+
+    return args
+
+
+def get_errant_ttl(db_client, args):
+    """
+    return all records with errant ttl in columnFamily
+    """
+    cqlstr = "SELECT key, column1, value, ttl(value) as ttl_val FROM " + args.columnFamily
+    if isVerbose: print "cqlstr: " + cqlstr
+    prepared_stmt = db_client.session.prepare(cqlstr)
+    results = db_client.session.execute(prepared_stmt)
+
+    errant_results = []
+    i = 0
+    for r in results:
+        if (r.ttl_val == None) or (int(r.ttl_val) > ERRANT_TTL_IN_SECONDS):
+            i += 1
+            if args.limit and i > int(args.limit): break
+            errant_results.append([r.key, r.column1, r.value, r.ttl_val])
+
+    return errant_results
+
+
+def update_errant_ttl(db_client, args, record):
+    """
+    update ttl of record by inserting new key,column1,value with new ttl
+    """
+    key = str(record[0])
+    column1 = str(record[1])
+    value = str(record[2])
+    cqlstr = "INSERT INTO %s (key, column1, value) values(%r, %s, %r) using ttl %s;" % (args.columnFamily, key, column1, value, str(ERRANT_TTL_IN_SECONDS))
```

Comment:
erk I tested it with insert and not update :p   But ok let me try it with update if this matters to you